cmd/snap-preseed/main.go

```diff
@@ -144,3 +182,23 @@ func run(parser *flags.Parser, args []string) (err error) {
 	}
 	return preseedClassic(chrootDir)
 }
+
+func downloadAssertion(assertType *asserts.AssertionType, headers map[string]string) (asserts.Assertion, error) {
```

Comment:
this should use the tooling store like cmd/snap does, we could even I suppose factor out some shared code in a package under cmd if it makes sense, or add more targeted helpers to store/tooling itself?